borgmatic/actions/restore.py

```diff
@@ -68,12 +68,15 @@ def restore_single_database(
     archive_name,
     hook_name,
     database,
+    schemas
 ):  # pragma: no cover
     '''
     Given (among other things) an archive name, a database hook name, and a configured database
     configuration dict, restore that database from the archive.
     '''
     logger.info(f'{repository}: Restoring database {database["name"]}')
+    if schemas:
+        database['schemas'] = schemas
```

Comment:
There's a risk that this mutates the database dict and "leaks" that mutation to callers of `restore_single_database`. Consider making a copy of `database` and mutating that instead. Alternatively, since it looks like the whole `database` dict is only used in a single place below, you could even do it inline:

```python
dict(database, **{'schema': schema})
```

(That syntax sidesteps duplicate keyword collisions on `schema`.)